sim/midas/src/main/scala/midas/widgets/PlusargsBridge.scala

```diff
@@ -0,0 +1,210 @@
+      s"name passed to PlusargsBridge (${params.name}) must contain =%d. Currently only the format %d is supported",
+    )
+
+    val target = Module(new PlusargsBridge(params))
+    target.io.clock := clock
+    target
+  }
+
+  /** apply overload which takes the parameters without the case class The signature was copied from
+    * chipyard/generators/rocket-chip/src/main/scala/util/PlusArg.scala
+    *
+    * @param name
+    *   string identifier, should include "name=%d"
+    *
+    * @param default
+    *   The value of the register if no matching plusarg is provided
+    *
+    * @param docstring
+    *   The doctring
+    *
+    * @param width
+    *   The width of the register, in bits
+    */
+  // Signature copied from chipyard/generators/rocket-chip/src/main/scala/util/PlusArg.scala
+  def apply(name: String, default: BigInt = 0, docstring: String = "", width: Int = 32): PlusargsBridge = {
+    val params = PlusargsBridgeParams(name, default, docstring, width)
+    annotatePlusargsBridge(Module.clock, Module.reset, params)
+  }
+
+  /** apply overload which takes the parameters case class
+    *
+    * @param params
+    *   Describes the name, width and default plusarg value
+    */
+  def apply(params: PlusargsBridgeParams): PlusargsBridge = {
+    annotatePlusargsBridge(Module.clock, Module.reset, params)
+  }
+
+  /** similar to apply, but named drive. This overload returns the out value. This overload is probably the one you want
+    *
+    * @param params
+    *   Describes the name, width and default plusarg value
+    */
+  def drive(params: PlusargsBridgeParams): UInt = {
+    val target = annotatePlusargsBridge(Module.clock, Module.reset, params)
+    target.io.out
+  }
+}
+
+/** The host-side implementation. Calculates widths and bundles multiple 32 bit MMIO to the exact width of
+  * `params.width`. If `params.width` does not evenly divide by 32 a MMIO with the remainder bits will be created. The
+  * remainders name will always be `outN` where `N` is the last MMIO
+  *
+  * `initDone` is used as valid for the outChannel. The C++ driver ensures that all `outN` are driven before `initDone`
+  * is asserted
+  *
+  * @param params
+  *   Describes the name, width and default plusarg value
+  */
+class PlusargsBridgeModule(params: PlusargsBridgeParams)(implicit p: Parameters)
+    extends BridgeModule[PlusargsBridgeHostIO]()(p) {
+  lazy val module = new BridgeModuleImp(this) {
+
+    val io    = IO(new WidgetIO())
+    val hPort = IO(new PlusargsBridgeHostIO(params)())
+
+    // divide with a ceiling round, to get the total number of slices
+    val sliceCount = (params.width + ctrlWidth - 1) / ctrlWidth;
+
+    // create a seq of widths of each slice
+    val slicesWidths = (0 until sliceCount).map(x => math.min(params.width - (x * ctrlWidth), ctrlWidth))
+
+    // zip/map the widths to call genWOReg
+    // reg names are out0, out1, ...
+    val slices = slicesWidths.zipWithIndex.map({ case (width, idx) =>
```

Comment:
I'm surprised the scalafmt doesn't check this but drop you should drop the parens when using these higher order functions. 
```suggestion
    val slices = slicesWidths.zipWithIndex.map { case (width, idx) =>
```